29 create chat view

benjaminpreiss opened the pull request.

benjaminpreiss committed:
WIP: Add chat view and view switching capabilities

This commit introduces a new chat view option in the social media app, alongside the existing thread view. It includes:

- New ViewProvider context to manage view state (chat/thread)
- Enhanced Reply and CanvasPreview components to support chat-message variant
- Added "chat" as a new sort criteria option in the dropdown
- Auto-switching between views based on sort criteria selection
- UI improvements for the chat view with appropriate styling

The changes allow users to toggle between thread and chat display modes for better conversation flow.

Note: This is work in progress and may need further UI refinements and testing.

Commit:
fix: generate posts with different identities

benjaminpreiss committed:
feat: enhance chat UI with improved message alignment and identity features

- Add left/right message alignment based on sender identity
- Improve chat bubble styling and container dimensions
- Implement proper avatar sizing in message headers
- Fix chat view sorting to display messages chronologically
- Add isMe utility function to identify user's messages across devices
- Fix height and spacing issues in conversation components

benjaminpreiss committed:
refactor(frontend): improve Preview and Reply components architecture

BREAKING CHANGE: Reply component API changed to use variant prop instead of separate view/variant props

- Refactor Preview component into smaller, focused subcomponents
- Improve type safety with discriminated union types for preview variants
- Restructure Reply component to use CSS grid for better layout control
- Add chat message alignment support with left/right positioning
- Extract blur background into separate reusable component
- Fix typo in separateAndSortRects function name
- Add utility for combining Tailwind classes
- Improve code organization and readability throughout

Commit:
fix: Merge branch 'master' into 29-create-chat-view

marcus-pousette commented on packages/social-media-app/frontend/src/identity/useIdentities.tsx:
```diff
@@ -54,12 +56,48 @@ export const IdentitiesProvider = ({ children }: { children: JSX.Element }) => {
+            return devices.some((connection) => {
+                return (
+                    Buffer.from(connection.device1.publicKey).equals(
+                        Buffer.from(keyBytes)
```
I am suprised how your editor works because Buffer.from(connection.device1.publicKey). should give Typescript compile errors. If you want to access the bytes of the key you can do

connection.device1.publicKey.bytes. 

But I am also curious what part of the chat view requires the isMe function

benjaminpreiss committed:
refactor: improve frontend presentation and types

- Add type safety to tailwind utility functions
- Optimize breadcrumb navigation layout and sizing
- Add JSDoc documentation to content components
- Clean up conditional class name assignments
- Add markdown parsing utilities for preview text

benjaminpreiss committed:
Merge branch '29-create-chat-view' of github.com:dao-xyz/peerbit-examples into 29-create-chat-view

benjaminpreiss committed:
fix: breadxrumb expanded dark mode

benjaminpreiss committed:
fix: expanded breadcrumb dark mode orange bg

benjaminpreiss committed:
Merge branch 'master' into 29-create-chat-view

benjaminpreiss committed:
fix: make breadcrumb bar always be content high

benjaminpreiss committed:
Merge branch 'master' into 29-create-chat-view
